server/src/test/java/org/apache/druid/metadata/SqlSegmentsMetadataManagerTest.java

```diff
@@ -265,7 +265,7 @@ public void testPollPeriodicallyAndOnDemandInterleave() throws Exception
     Assert.assertTrue(sqlSegmentsMetadataManager.getLatestDatabasePoll() instanceof SqlSegmentsMetadataManager.PeriodicDatabasePoll);
     dataSourcesSnapshot = sqlSegmentsMetadataManager.getDataSourcesSnapshot();
     Assert.assertEquals(
-        ImmutableList.of("wikipedia2", "wikipedia3", "wikipedia"),
+        ImmutableList.of("wikipedia3", "wikipedia", "wikipedia2"),
```

Comment:
why does this order change?

Reply:
Changes to this method `com.google.common.collect.Maps.newHashMapWithExpectedSize(int expectedSize)` which we're calling from here https://github.com/apache/druid/blob/a8eaa1e4ed81f94fe53ae14bbb078678e35de105/processing/src/main/java/org/apache/druid/utils/CollectionUtils.java#L103 caused this order to change

Reply:
looks like the order changed again in https://github.com/apache/druid/pull/15482/files#r1414283468. Given that we are not order sensitive it might make more sense to compare sets.